Libraries/Opc.Ua.Server/Diagnostics/MonitoredNode.cs

```diff
@@ -234,18 +234,18 @@ public void OnReportEvent(ISystemContext context, NodeState node, IFilterTarget
                         else
                         {
                             // check if channel is not encrypted and skip if so
-                            OperationContext operationContext = (context as SystemContext)?.OperationContext as OperationContext;
-                            if (operationContext != null &&
-                                operationContext.ChannelContext.EndpointDescription.SecurityMode != MessageSecurityMode.SignAndEncrypt &&
-                                operationContext.ChannelContext.EndpointDescription.TransportProfileUri != Profiles.HttpsBinaryTransport)
+                            if (monitoredItem?.Session?.EndpointDescription?.SecurityMode != MessageSecurityMode.SignAndEncrypt &&
```

Comment:
I'd also not pass the audit event if Session is undefined? Using ? may fall through

Reply:
or does it compare null != signadencrypt?

Reply:
compares null != signandencrypt

Reply:
compares null != signadencrypt